### ISSUE TITLE
[Feature Request] Adding a way to add keys automatically in all layers if added on defsrc

### ISSUE THREAD

**Issue by jvlara** [rszyma/vscode-kanata#39]
Hi, first of all, thank you for this extension; it is really handy!

I have a large configuration file in my kdb file with 7 layers, and now I want to add more keys. However, adding them in the same order across every layer is a bit of a pain.

If the autoformatter could detect when I add something to defsrc and automatically add _ on every defined layer, that would be really useful. The potential issue I see is that this behavior might not be intended if process-unmapped-keys is set to no, which is not my case.

What do you think? Does it make sense?

**Comment by rszyma** [comment]
Makes sense. I'm can't think of the reason why you say process-unmapped-keys would interfere with this feature, can you explain?

Detailed behavior I would implement: 
- only adding (1 or more) keys to defsrc should trigger adding _ to all deflayers; removing won't (unless all deflayers in a particular position contain _)
- number of keys in all deflayers must be the same to trigger this

Potential problems:
- the formatter is currently stateless (doesn't have info about previous state), so adding a new key to the end of defsrc will work without issues, but adding a new key in the middle, the formatter will have no idea it was added to a specific position, only that "there's now 1 more key than all other deflayers)

**Comment by jvlara** [comment]
Thinking about this a second time makes me realize that process-unmapped-keys should not be affected by this. I didn't go into the details of the code, but when formatting, I think it would be possible to scan the file and detect that defsrc has one key that the other does not and where that key is
